src/main/java/com/google/opensesame/projects/ProjectEntity.java

```diff
@@ -84,33 +65,28 @@ public ProjectEntity(
     }
   }
 
-  public int getNumMentors() {
-    return this.mentorIds.size();
+  /** Compute the lengths of mentor and interested user lists before the project entity is saved. */
+  @OnSave
+  protected void computeListLengths() {
+    numMentors = mentorIds.size();
+    numInterestedUsers = interestedUserIds.size();
   }
 
-  public int getNumInterestedUsers() {
-    return this.interestedUserIds.size();
+  @OnLoad
+  protected void checkNullLists() {
+    if (this.mentorIds == null) {
+      this.mentorIds = new ArrayList<String>();
+    }
+    if (this.interestedUserIds == null) {
+      this.interestedUserIds = new ArrayList<String>();
+    }
   }
 
-  /**
-   * Adds the properties of this ProjectEntity object to an Entity.
-   *
-   * @param entity The Datastore Entity to add this ProjectEntity's properties to.
-   */
-  public void fillEntity(Entity entity) {
-    entity.setProperty(REPOSITORY_ID_KEY, repositoryId);
-    entity.setProperty(MENTOR_IDS_KEY, mentorIds);
-    entity.setProperty(INTERESTED_USER_IDS_KEY, interestedUserIds);
-    entity.setProperty(NUM_MENTORS_KEY, getNumMentors());
-    entity.setProperty(NUM_INTERESTED_USERS_KEY, getNumInterestedUsers());
+  public int getNumMentors() {
+    return this.mentorIds.size();
```

Comment:
Are this function and `getNumInterestedUsers` used anywhere? If not, maybe we can skip them for now.

I'm not 100% sure that they're better than just having the user call `ProjectEntity.mentordIds.size()` and there's no use-case currently.

Reply:
That's a good point. `getNumMentors()` is currently being used in [`ProjectPreviewData.fromProjectEntity()`](https://github.com/googleinterns/step22-2020/blob/a912657e1f7f3b54f324e0b2810e16d544292e1f/src/main/java/com/google/opensesame/projects/ProjectPreviewData.java#L37) but I'm surprised I didn't notice how unnecessary these are because the lists themselves are public. I'll remove these.